builds/Makefile

```diff
@@ -1885,238 +1479,61 @@ $(WATCH_MONO_PATH)/tools/offsets-tool/MonoAotOffsetsDumper.exe: $(wildcard $(WAT
-# parameters are:
-#  $(1): target ABI
-#  $(2): output directory
-#  $(3): build
-define GenerateCrossOffsets
-$(3)/$(1).h: .stamp-configure-$(3) target7/mono/arch/arm/arm_dpimacros.h targetwatch/mono/arch/arm/arm_dpimacros.h $(4)/tools/offsets-tool/MonoAotOffsetsDumper.exe
-	$(Q) MONO_PATH=$(4)/tools/offsets-tool/CppSharp/osx_32 \
-	$(SYSTEM_MONO32) $(4)/tools/offsets-tool/MonoAotOffsetsDumper.exe --abi $(1) --out $(2) --mono $(abspath $(4)) --maccore $(abspath $(TOP))
-endef
-
-CROSS_BASE_CFLAGS=-DMONOTOUCH $(CROSS_BUILD_CFLAGS) $(CCACHE_CFLAGS)
-CROSS_BASE_CXXFLAGS=-DMONOTOUCH $(CCACHE_CXXFLAGS)
-
-CROSS_BASE_CONFIGURE_FLAGS= \
-			--with-glib=embedded \
-			--enable-minimal=com,remoting \
-			--disable-mcs-build \
-			--enable-llvm \
-			--enable-icall-symbol-map \
-			--disable-nls \
-			--disable-iconv \
-			--disable-libraries	\
-			--disable-boehm \
-			$(XAMARIN_IOS_CONFIGURE_FLAGS)
-
-#
-# 32 bit cross compiler
-#
-
-# mono's configure already adds an -arch i386 flag, and cache doesn't like duplicate -arch flags
-CROSS_CFLAGS=$(CROSS_BASE_CFLAGS) -mmacosx-version-min=$(MIN_OSX_SDK_VERSION)
-CROSS_CXXFLAGS=$(CROSS_BASE_CXXFLAGS) -mmacosx-version-min=$(MIN_OSX_SDK_VERSION) -stdlib=libc++
-CROSS_LDFLAGS=-stdlib=libc++ $(COMMON_LDFLAGS)
-
-CROSS_CONFIGURE_FLAGS=$(CROSS_BASE_CONFIGURE_FLAGS)	\
-	--prefix=$(BUILD_DESTDIR)/cross \
-	--build=i386-apple-darwin10 \
-	--cache-file=../cross.config.cache \
-	--target=arm-darwin \
-	--with-cross-offsets=arm-apple-darwin10.h \
-	--with-llvm=../llvm/usr
-
-CROSS_CONFIGURE_ENVIRONMENT = \
-	CFLAGS="$(CROSS_CFLAGS)" \
-	CXXFLAGS="$(CROSS_CXXFLAGS)" \
-	LDFLAGS="$(CROSS_LDFLAGS)" \
-	CC="$(MAC_CC)" \
-	CXX="$(MAC_CXX)" \
-
-ifdef INCLUDE_IOS
-ifdef INCLUDE_DEVICE
-cross:: build-cross install-cross
-
-.PHONY: setup-cross build-cross install-cross
-
-setup:: setup-cross
-build:: build-cross
-install-local:: install-cross
-clean-local:: clean-cross
-endif
-endif
-
-setup-cross: .stamp-configure-cross
-
-.stamp-configure-cross: $(MONO_PATH)/configure .stamp-build-llvm
-	$(Q) $(CROSS_CONFIGURE_ENVIRONMENT) ./wrap-configure.sh cross $(abspath $(MONO_PATH)/configure) $(CROSS_CONFIGURE_FLAGS)
-
-$(eval $(call GenerateCrossOffsets,arm-apple-darwin10,$(TOP)/builds/cross/,cross,$(MONO_PATH)))
-
-.stamp-build-cross: .stamp-configure-cross cross/arm-apple-darwin10.h $(MONO_DEPENDENCIES)
-	$(MAKE) -C cross/mono
-	$(MAKE) -C cross/mono install
-	$(Q) touch $@
-
-build-cross: .stamp-build-cross
-
-CROSS_TARGETS = \
-	$(PREFIX)/bin/arm-darwin-mono-sgen  \
-
-$(BUILD_DESTDIR)/cross/bin/arm-darwin-mono-sgen: .stamp-build-cross
-
-$(PREFIX)/bin/arm-darwin-mono-sgen: $(BUILD_DESTDIR)/cross/bin/arm-darwin-mono-sgen | $(PREFIX)/bin
-	$(call Q_2,INSTALL ,[CROSS]) install -c -m 0755 $(INSTALL_STRIP_FLAG) $< $@
-
 $(PREFIX)/bin:
 	$(Q) mkdir -p $@
 
-install-cross: install-llvm64 $(CROSS_TARGETS)
+# Dependency used to build the offsets tool before recursively calling sdk makefiles
+offsets-tool: $(MONO_PATH)/configure
```

Comment:
I think this can just depend on `$(MONO_PATH)/tools/offsets-tool/MonoAotOffsetsDumper.exe` instead:

```make
offsets-tool: $(MONO_PATH)/tools/offsets-tool/MonoAotOffsetsDumper.exe
```

and then it will be built when needed.